pkg/gadgets/run/tracer/tracer.go

```diff
@@ -860,6 +924,11 @@ func (t *Tracer) Run(gadgetCtx gadgets.GadgetContext) error {
 }
 
 func (t *Tracer) AttachContainer(container *containercollection.Container) error {
+	// Only attach to containers if ifaceName is not set
+	if t.ifaceName != "" {
+		return nil
+	}
```

Comment:
Why here? `SocketFilter` programs doesn't have the iface parameter so they shouldn't by affected by this.